tools/preflight/helper.go

```diff
@@ -180,6 +186,9 @@ func GetHelmVersion() (string, error) {
 }
 
 func GetServerPreferredVersionForGroup(grp string, cl *goclient.Clientset) (string, error) {
+	if cl == nil {
+		return "", fmt.Errorf("client object is nil, cannot fetch versions of group - %s", grp)
+	}
```

Comment:
add this condition where you have initialized this object... no need to check it in every func.
Remove such checks from all utility funcs and remove all test-cases related to this nil check.